src/main/java/world/bentobox/aoneblock/listeners/BlockListener.java

```diff
@@ -343,7 +343,11 @@ private void makeSpace(Entity e) {
                 for (; y <= Math.min(bb.getMaxY() + 1, world.getMaxHeight()); y++) {
                     b = world.getBlockAt(new Location(world, x,y,z));
                     if (!b.getType().equals(Material.AIR) && !b.isLiquid()) b.breakNaturally();
-                    b.setType(WATER_ENTITIES.contains(e.getType()) ? Material.WATER : Material.AIR, false);
+                    if (addon.getSettings().isWaterMobProtection()) {
+                        b.setType(WATER_ENTITIES.contains(e.getType()) ? Material.WATER : Material.AIR, false);
```

Comment:
Replace lines 346 to 350 with this one line:

```
b.setType(WATER_ENTITIES.contains(e.getType()) && addon.getSettings().isWaterMobProtection() ? Material.WATER : Material.AIR, false);
```

Reply:
This indeed is better. I changed it, thanks.